notebooks/features/isolation_forest/IsolationForest - Multivariate Anomaly Detection.ipynb

```diff
@@ -158,8 +147,65 @@
     "# MLFlow experiment\n",
     "artifact_path = \"isolationforest\"\n",
     "experiment_name = f\"/Shared/isolation_forest_experiment-{str(uuid.uuid1())}/\"\n",
-    "model_name = \"isolation-forest-model\"\n",
-    "model_version = 1"
+    "model_name = f\"isolation-forest-model-{str(uuid.uuid1())}\"\n",
+    "model_version = 1\n",
+    "if running_on_synapse():\n",
+    "    import subprocess\n",
+    "    import sys\n",
+    "    import os\n",
+    "    from synapse.ml.core.platform import find_secret\n",
+    "    from pyspark.sql.functions import udf\n",
+    "    from notebookutils.visualization import display\n",
+    "\n",
+    "    # from synapse.ml.core.platform import materializing_display as display\n",
+    "    os.environ[\"AZURE_TENANT_ID\"] = find_secret(\"sp-aml-azure-tenant-id\")\n",
+    "    os.environ[\"AZURE_CLIENT_ID\"] = find_secret(\"sp-aml-azure-client-id\")\n",
+    "    os.environ[\"AZURE_CLIENT_SECRET\"] = find_secret(\"sp-aml-azure-client-secret\")\n",
+    "\n",
+    "    # solve version conflict, typing_extensions is already loaded into the namespace for the ML runtime, it does not get reloaded. Manually trigger reload with:\n",
+    "    subprocess.check_call(\n",
+    "        [sys.executable, \"-m\", \"pip\", \"install\", \"typing_extensions==4.4.0\"]\n",
+    "    )  # only needed for spark 3.2 pool\n",
+    "    import typing_extensions  # only needed for spark 3.2 pool\n",
+    "    from importlib import reload  # only needed for spark 3.2 pool\n",
+    "\n",
+    "    reload(typing_extensions)  # only needed for spark 3.2 pool\n",
+    "\n",
+    "    subprocess.check_call([sys.executable, \"-m\", \"pip\", \"install\", \"azure.ai.ml\"])\n",
+    "    subprocess.check_call([sys.executable, \"-m\", \"pip\", \"install\", \"sqlparse\"])\n",
+    "    subprocess.check_call([sys.executable, \"-m\", \"pip\", \"install\", \"raiwidgets\"])\n",
+    "    subprocess.check_call(\n",
+    "        [sys.executable, \"-m\", \"pip\", \"install\", \"interpret-community\"]\n",
+    "    )\n",
+    "\n",
```

Comment:
this looks a bit hairy, perhaps modify the script that submits jobs here

Reply:
updated, please review.